injection/sharedmain/main.go

```diff
@@ -111,6 +113,38 @@ var (
 	WebhookMainWithConfig  = MainWithConfig
 )
 
+// MainNamed runs the generic main flow for controllers and webhooks.
+func MainNamed(ctx context.Context, component string, ctors ...injection.NamedControllerConstructor) {
+
+	disabledControllers := flag.String("disable-controllers", "", "Comma-separated list of disabled controllers.")
```

Comment:
can we use [pflag.StringSlice](https://pkg.go.dev/github.com/spf13/pflag#StringSlice) here?